### PR TITLE
Test scripts only run cases defined in the running script

### PR DESCRIPTION
**Pull request by ssnl**
1. Refactors `TestTorch` into `TestTorchMixin` (subclass of `object`) and `TestTorch` (subclass of `TestCase`, MRO `(TestCase, TestTorchMixin)`, only defined if `__name__ == '__main__'`). So other scripts won't accidentally run it.
2. Adds an assertion in `load_tests` that each script only runs cases defined in itself.

cc @yf225 @ezyang

### CODE REVIEW

**Comment by ezyang** [comment]
Ooh, thank you!!!

File: test/common_utils.py [hunk 1]
```diff
@@ -779,16 +779,40 @@ def get_int64_dtype(dtype):
...
 }
 
+
+running_script_path = None
+
+
+def set_running_file():
+    global running_script_path
+    try:
+        running_script_path = os.path.abspath(os.path.realpath(sys.argv[0]))
```
Comment (ezyang) [inline test/common_utils.py]: Does this work in fbcode?

**Review by ezyang — APPROVED** [review]
Can you check that the sanity checking code works in fbcode? Thanks!

File: test/common_utils.py (hunk 1 above)
Comment (SsnL) [reply test/common_utils.py]: ahh good question. I'll try.

**Comment by vishwakftw** [comment]
@SsnL I think the changes in this PR (refactoring `TestTorch` into `TestTorchMixin` and `TestCase`) is causing issues with `pytest`. Earlier, I was able to run specific tests by passing the argument `-k <name_of_test>`, but now it doesn't seem to work.

**Comment by vishwakftw** [comment]
No worries, I thought I'd put it here instead of creating an issue, but seems like @zou3519 did it.

**Pull request closed without merging**